deepinv/datasets/datagenerator.py

```diff
@@ -58,69 +95,168 @@ def __init__(
         dtype: torch.dtype = torch.float,
         complex_dtype: torch.dtype = torch.cfloat,
     ):
+        import h5py
+
         super().__init__()
-        self.data_info = []
-        self.data_cache = {}
-        self.unsupervised = False
+
+        f = h5py.File(path, "r")
+
+        self._register_members(
```

Comment:
Again, is there a reason why this is a separate method rather than in-line or is it stylistic?